### DISCUSSION TITLE
m.add_time_slider leafmap do not work in streamlit

### DISCUSSION THREAD

**Discussion by HibaDoi** [opengeos/leafmap#626] [Q&A] [answered]
I want to create a time slider from tiff images in my directory  into a Streamlit app:  and i have two problems: the first one is that 
==> The folium_static function does not allow this map visualization despite the same code is working successfully in collab 
==> The second one is that there is not a way to import my layer without using Planet or stac ....
###############################################################################
from leafmap import leafmap
from streamlit_folium import folium_static,st_folium

import ipyleaflet
import json
import requests

stac_api = "https://earth-search.aws.element84.com/v0"
search_endpoint = f"{stac_api}/search"

collection = "sentinel-s2-l2a-cogs"
payload = {
    "bbox": [
        -102.83340454101562,
        49.77860375256143,
        -102.41043090820312,
        50.05273014900257,
    ],
    "datetime": "2021-07-01T00:00:00Z/2021-10-01T12:31:12Z",
    "collections": [collection],
    "limit": 10,
    "query": {"eo:cloud_cover": {"gte": 0, "lte": 10}},
}

headers = {'Content-Type': 'application/json'}

response = requests.request(
    "POST", search_endpoint, headers=headers, data=json.dumps(payload)
)

features = response.json()["features"]
features.sort(key=lambda x: x["properties"]["datetime"], reverse=False)

layers_dict = {}
for feature in features:
    feature_id = feature["id"]
    print(feature_id)

    url = leafmap.stac_tile(
        f"{stac_api}/collections/{collection}/items/{feature_id}", bands=["visual"]
    )
    tile_layer = ipyleaflet.TileLayer(
        url=url,
        name=feature_id,
    )
    layers_dict[feature_id] = tile_layer

m = leafmap.Map(center=[50.093079, -103.152825], zoom=11)
m.add_time_slider(layers_dict, time_interval=2)
st_folium(m)

**Comment by giswqs** [comment, marked as answer]
Streamlit doesn't support ipywidgets. The time slider will never work with streamlit. Try solara https://github.com/opengeos/solara-geospatial

**Reply by HibaDoi** [reply]
from leafmap import leafmap

m = leafmap.Map()

m.add_time_slider(["/content/pr_3cpp.tif","/content/pr_4cpp.tif"])

m

i just tried this code and it is working in colab ; still no hope of working in streamlit despite not using ipywidgets?

**Reply by giswqs** [reply]
Because colab support ipywidgets, but streamlit doesn't. Don't waste your time.

**Reply by HibaDoi** [reply]
I have another question: could this be the reason why it's not functioning properly with the Larboard library?
